vertx-web-api-contract/src/main/java/io/vertx/ext/web/api/contract/openapi3/impl/OpenAPI3RouterFactoryImpl.java

```diff
@@ -168,12 +184,18 @@ public OpenAPI3RouterFactoryImpl(Vertx vertx, OpenAPI spec, ResolverCache refsCa
     /* --- Initialization of all arrays and maps --- */
     for (Map.Entry<String, ? extends PathItem> pathEntry : spec.getPaths().entrySet()) {
       for (Map.Entry<PathItem.HttpMethod, ? extends Operation> opEntry : pathEntry.getValue().readOperationsMap().entrySet()) {
-        this.operations.put(opEntry.getValue().getOperationId(), new OperationValue(
-          HttpMethod.valueOf(opEntry.getKey().name()),
-          pathEntry.getKey(),
-          opEntry.getValue(),
-          pathEntry.getValue()
-        ));
+          if (opEntry.getValue().getOperationId() != null) {
+              this.operations.put(opEntry.getValue().getOperationId(), 
+                      new OperationValue(
+                              HttpMethod.valueOf(opEntry.getKey().name()),
+                              pathEntry.getKey(),
+                              opEntry.getValue(),
+                              pathEntry.getValue()
+                          ));
+          } else {
+              LOG.warn("The operation '"+ opEntry.getKey() + "' from path '" + 
+                      pathEntry.getKey()+"' is being ignored because it doesn't have an operationId");
```

Comment:
```suggestion
              LOG.warn("The operation '{}' from path '{}' is being ignored because it doesn't have an operationId", opEntry.getKey(), pathEntry.getKey());
```
If logging level is set to error (by the way, maybe, using LOG.error is worth considering here?), the string concatenation will not be done and the message simply discarded. 
Using `+` concatenates no matter the log level.

It's a very small perf. improvement, but hopefully a good practice that'll prove useful for you in the future :)

Reply:
@aesteve, I used to use SLF4J, so I always would agree with your proposal ;)
But I cannot find parameterized methods. I read in the dev groups and confirmed that they were removed due to their lack of portability. 

So, I would agree to include an "if (LOG.isWarnEnabled()) "